extensions/ql-vscode/src/view/model-editor/ModelKindDropdown.tsx

```diff
@@ -63,7 +63,16 @@ export const ModelKindDropdown = ({
 
   useEffect(() => {
     const value = modeledMethod?.kind;
-    if (value === undefined && kinds.length > 0) {
+
+    if (kinds.length === 0) {
+      if (value !== "") {
+        onChangeKind("");
```

Comment:
I've tested it and this PR works. It took me a little while to understand why.

What confused me is that in `onChangeKind` it aborts and does nothing if `modeledMethod === undefined`, so in the case of there being no modeled method, this PR shouldn't be a change in behaviour. And in the case where there is a modeled method but `type === "none"`, then the according the typing, `kind` should never be undefined.

But the issue isn't with `value` being undefined to start with, and instead it was in `value !== undefined && !kinds.includes(value)` where the `kinds` array would be empty so we'd call `onChangeKind(undefined)`.

This PR fixes things by adding a check for `kinds.length === 0` earlier.

Looking into this made me realise that we no longer call these inputs with `modeledMethod: undefined` in the model editor panel. In `MethodRow` we always construct a dummy modeled method with `type: "none"`.